### PR TITLE
Add support for services: quota-status, quota-warning, doveadm

### PR DESCRIPTION
**Pull request by ledgr**
Service description examples:

```ruby
default['dovecot']['services']['doveadm'] = {
  'listeners' => [
    { 'inet:doveadm-server' => { 'port' => 3333 }
    }
  ]
}

default['dovecot']['services']['quota-status'] = {
  'executable' => 'quota-status -p postfix',
  'listeners' => [
    { 'inet:imap' => { 'port' => 4444 } }
  ]
}

default['dovecot']['services']['quota-warning'] = {
  'user' => 'dovecot',
  'executable' => 'script /usr/local/bin/quota-warning.sh',
  'listeners' => [
    { 'unix:quota-warning' => { 'user' => 'postfix' } }
  ]
}
```

### CODE REVIEW

File: templates/default/conf.d/90-quota.conf.erb [hunk 1]
```diff
@@ -7,8 +7,16 @@
...
+<% if @plugins.has_key?('quota') or @services['quota-status'].kind_of?(Hash) or @services['quota-warning'].kind_of?(Hash) -%>
+  <% if @plugins['quota'].kind_of?(Hash) -%>
 <%=  Dovecot::Conf.plugin('quota', @plugins['quota']) %>
+  <% end -%>
+  <% if @services['quota-status'].length > 0 -%>
+<%=  Dovecot::Conf.service('quota-status', @services['quota-status']) %>
+  <% end -%>
+  <% if @services['quota-warning'].length > 0 -%>
+<%=  Dovecot::Conf.service('quota-warning', @services['quota-warning']) %>
+  <% end -%>
```
Comment (zuazo) [inline templates/default/conf.d/90-quota.conf.erb]: I think we should homogenize the conditionals here, shouldn't we?

Maybe something like the following:

```erb
<% if @plugins['quota'].kind_of?(Hash) or @services['quota-status'].kind_of?(Hash) or @services['quota-warning'].kind_of?(Hash) -%>
  <% unless @plugins['quota'].empty? -%>
    [...]
  <% end -%>
  <% unless @plugins['quota-status'].empty? -%>
    [...]
  <% end -%>
  <% unless @plugins['quota-warning'].empty? -%>
    [...]
  <% end -%>
<% else -%>
```

**Comment by zuazo** [comment]
Hi Edgaras,

Thanks for your PR!

In addition to what I commented on the code (a tiny detail), it would be great to add those usage examples in the README :wink:

File: templates/default/conf.d/90-quota.conf.erb (hunk 1 above)
Comment (ledgr) [inline templates/default/conf.d/90-quota.conf.erb]: I have pushed my second go. Checking empty?, kind_of? or anything else on NilClass will throw error. I'm not rubyist so I hope my second go is ok :)

**Comment by ledgr** [comment]
Updated README

**Comment by zuazo** [comment]
It's perfect now :smiley: Thank you!

**Comment by zuazo** [comment]
Released in [`3.0.0`](https://github.com/zuazo/dovecot-cookbook/releases/tag/3.0.0).

**Pull request merged**